### PR TITLE
Gli/refactor

### PR DESCRIPTION
**Pull request by glusa8**

### CODE REVIEW

File: airship_python/__init__.py [hunk 1]
```diff
@@ -124,7 +124,7 @@
 GATING_INFO_ENDPOINT = '{}/v1/gating-info'.format(SERVER_URL)
 PLATFORM = 'python'
 
-SDK_VERSION = '{}:{}'.format(PLATFORM, '1.1.9')
+SDK_VERSION = '{}:{}'.format(PLATFORM, '1.1.10')
```
Comment (glusa8) [inline airship_python/__init__.py]: lgtm

File: airship_python/__init__.py [hunk 2]
```diff
@@ -162,62 +162,52 @@ class Airship:
     SDK_ID = ''.join([choice(ascii_lowercase + ascii_uppercase + digits) for _ in range(6)])
 
     def __init__(self, options):
-
         self.api_key = options.get('api_key')
         self.env_key = options.get('env_key')
+        self.timeout = 10
```
Comment (glusa8) [inline airship_python/__init__.py]: lgtm

File: airship_python/__init__.py [hunk 3]
```diff
@@ -162,62 +162,52 @@ class Airship:
...
 
-        if self.api_key is None:
-            raise Exception('Missing api_key')
+        self.polling_interval = 60
+        self.ingestion_interval = 60
```
Comment (glusa8) [inline airship_python/__init__.py]: lgtm

File: airship_python/__init__.py (hunk 2 above)
Comment (libcxx) [reply airship_python/__init__.py]: 👍

File: airship_python/__init__.py [hunk 4]
```diff
@@ -162,62 +162,52 @@ class Airship:
...
-        self.gate_stats_upload_batch_interval = 60
-
-        self.gate_stats_watcher = None
-        self.gate_stats_last_task_scheduled_timestamp = 0
+        self.last_ingestion_timestamp = 0
+        self.ingestion_tasks = []
 
-        self.gate_stats_uploader_tasks = []
+        self.ingestion_batch = []
+        self.ingestion_max_batch_size = 500
```
Comment (libcxx) [inline airship_python/__init__.py]: Nitpick: Would it be a good idea to put these numeric constants into global variables? Something like:

```
DEFAULT_INGESTION_MAX_BATCH_SIZE=500
self.ingestion_max_batch_size = DEFAULT_INGESTION_MAX_BATCH_SIZE
```

File: airship_python/__init__.py [hunk 5]
```diff
@@ -318,33 +318,34 @@ def _process_batch(self, limit, gate_stats=None):
...
             processed = True
 
-        self.gate_stats_batch_lock.release()
+        self.ingestion_batch_lock.release()
         return processed
 
-    def _upload_stats_async(self, gate_stats):
-        processed = self._process_batch(self.max_gate_stats_batch_size - 1, gate_stats)
+    def _ingest_async(self, gate_stats):
+        processed = self._process_batch(self.ingestion_max_batch_size - 1, gate_stats)
```
Comment (libcxx) [inline airship_python/__init__.py]: Would it ever be possible for the value passed to `_process_batch` to go below 0?

File: airship_python/__init__.py (hunk 4 above)
Comment (glusa8) [reply airship_python/__init__.py]: Great point. Thanks @libcxx , I will consider that in the next refactor!

File: airship_python/__init__.py (hunk 5 above)
Comment (glusa8) [reply airship_python/__init__.py]: Right now, we control what the `ingestion_max_batch_size` is, so no, but good practice to do input validation.

**Pull request merged**